### ISSUE TITLE
[Question] Integrate something like jaxtyping for more specific type annotations and runtime type-checking

### ISSUE THREAD

**Issue by dantp-ai** [araffin/sbx#63]
### Question

Are there any plans on the roadmap to use something like [jaxtyping](https://github.com/patrick-kidger/jaxtyping) for more specific type annotations and runtime type checking?

### Checklist

- [x] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/) (**required**)
- [x] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo (**required**)

**Comment by araffin** [comment]
Hello,
there is currently no plan for it.
Would you volunteer for such feature?

**Comment by dantp-ai** [comment]
I can look into it. I'll start first with the tests folder.

**Comment by sandeshkatakam** [comment]
Hey, 
I was actually considering opening an issue to explore adding jaxtyping support as well. It sounds like we're aligned in our interests, and If you like, I’d love to collaborate with you on this!

> I can look into it. I'll start first with the tests folder.

**Comment by dantp-ai** [comment]
@sandeshkatakam I'd be glad to collaborate. How should we do it best?

One idea would be that, I open a feature branch on my fork. You'll need to add my fork as a remote and check out the same branch. Once this is done, we can sync frequently and decided who works on specific parts to avoid conflicts. What do you think?

**Comment by sandeshkatakam** [comment]
> @sandeshkatakam I'd be glad to collaborate. How should we do it best?
> 
> One idea would be that, I open a feature branch on my fork. You'll need to add my fork as a remote and check out the same branch. Once this is done, we can sync frequently and decided who works on specific parts to avoid conflicts. What do you think?

Yeah this seems great! I will add your fork as remote and push commits there after we decide on dividing the specific parts of the work. Thank you :)

**Comment by sandeshkatakam** [comment]
> @sandeshkatakam I'd be glad to collaborate. How should we do it best?
> 
> One idea would be that, I open a feature branch on my fork. You'll need to add my fork as a remote and check out the same branch. Once this is done, we can sync frequently and decided who works on specific parts to avoid conflicts. What do you think?

Let me know once you fork the repo and create a branch

**Comment by dantp-ai** [comment]
@sandeshkatakam I'll open the PR at the latest on Thu.

**Comment by sandeshkatakam** [comment]
> @sandeshkatakam I'll open the PR at the latest on Thu.

hey did you open the PR??

**Comment by dantp-ai** [comment]
@sandeshkatakam I looked in a bit more detail how to use `jaxtyping` in the project. It can be paired with `mypy` for static analysis and additionally using e.g. `beartype` for runtime checks in critical sections (and avoid these runtime checks in performance-critical paths).

I'm blocked this week with some other stuff. Feel free to open a PR (from your fork) if you want to dive in already.
